extensions/bitwarden/src/utils.ts

```diff
@@ -44,36 +44,32 @@ export function getPasswordGeneratingArgs(options: PasswordGeneratorOptions): st
 
 export const capitalise = (value: string) => value.charAt(0).toUpperCase() + value.slice(1);
 
-export function withKeywords(items: Item[]): Item[] {
-  return items.map((item: Item) => {
-    const keywords = [item.name];
-    if (item.card) {
-      const { brand, number } = item.card;
-      if (brand !== null) {
-        keywords.push(brand);
-      }
-      if (number !== null) {
-        // Similar to Bitwarden, use the last 5 digits if the card is Amex
-        const isAmex = /^3[47]/.test(number);
-        keywords.push(number.substring(number.length - (isAmex ? 5 : 4), number.length));
-      }
-    }
-    if (item.login?.username) {
-      keywords.push(item.login.username);
+export function extractKeywords(item: Item): string[] {
+  const keywords: (string | null | undefined)[] = [item.name];
+  if (item.card) {
+    const { brand, number } = item.card;
+    keywords.push(brand);
+    if (number !== null) {
+      // Similar to Bitwarden, use the last 5 digits if the card is Amex
+      const isAmex = /^3[47]/.test(number);
+      keywords.push(number.substring(number.length - (isAmex ? 5 : 4), number.length));
     }
-    if (item.login?.uris) {
-      for (const uri of item.login.uris) {
-        if (uri.uri !== null) {
-          try {
-            keywords.push(...new URL(uri.uri).hostname.split("."));
-          } catch (error) {
-            // Invalid hostname
-          }
+  }
+  keywords.push(item.login?.username);
+  if (item.login?.uris) {
+    for (const uri of item.login.uris) {
+      if (uri.uri !== null) {
+        try {
+          keywords.push(...new URL(uri.uri).hostname.split("."));
+        } catch (error) {
+          // Invalid hostname
         }
       }
     }
-    // Unique keywords
-    item.keywords = [...new Set(keywords)];
-    return item;
-  });
+  }
+  // Unique keywords
```

Comment:
Looks good ! You can use `!!keyword` instead of `!(keyword === null || keyword === undefined)`

Reply:
Good catch, habit from avoiding resolving false for an empty string 😄 We don't want an empty string here so this is better. Cheers